api/src/main/java/marquez/db/DatasetDao.java

```diff
@@ -292,15 +292,27 @@ DatasetRow upsert(
       String name,
       String physicalName);
 
+  @SqlUpdate(
+      """
+    UPDATE datasets d
+    SET is_hidden = true
+    FROM dataset_symlinks ds
+    JOIN namespaces n
+    ON n.uuid=ds.namespace_uuid
+    WHERE ds.dataset_uuid = d.uuid
+    AND n.name=:namespaceName
+  """)
```

Comment:
Shouldn't we also consider namespace name of the symlink?

Reply:
Discussed offline. No - because the query captures other symlinks directly, and those are symlinks, not hard links - we don't want to hide them by design.